presto-main/src/main/java/com/facebook/presto/operator/aggregation/GenericAccumulatorFactory.java

```diff
@@ -151,10 +166,33 @@ public Accumulator createIntermediateAccumulator()
 
     @Override
     public GroupedAccumulator createGroupedAccumulator()
+    {
+        return spillEnabled ? createSpillableGroupedAccumulator() : createDefaultGroupedAccumulator();
+    }
+
+    @Override
+    public GroupedAccumulator createGroupedIntermediateAccumulator()
+    {
+        return spillEnabled ? createSpillableGroupedIntermediateAccumulator() : createDefaultGroupedIntermediateAccumulator();
```

Comment:
We do need this since before:
1. `createGroupedIntermediateAccumulator` used to be **never called** since order by and distinct did not have support for intermediate results. Hence, when spill is enabled, we use `createGroupedIntermediateAccumulator` to recreate the intermediate version of the accumulator with the spillable wrapper when unspilling and creating intermediate accumulators.

2. `checkState(!hasDistinct() || !hasOrderBy())` not sure what this does exactly in this context since having distinct AND orderby shouldnt be a state failure. `createGroupedIntermediateAccumulator` can be called with hasDistinct/hasOrderBy set to true.

3. `return groupedAccumulatorConstructor.newInstance` would be a bug since this references the underlying accumulator (NOT order by or distinct, but the accumulator inside of order by or distinct). This means that during intermediate accumulation we would get non-distinct  non-ordered values.

I actually tried writing this code without this section in a previous iteration but I realized we need the spillable wrapper in the intermediate case.